packages/dashboard-message-bus-common/lib/messages.ts

```diff
@@ -11,6 +9,20 @@ export interface Response {
   payload: any;
 }
 
+export type CliEventMessageType = "cli-event";
+export const cliEventMessageType = "cli-event";
+/**
+ * Message to inform subscribers of Truffle CLI event data.
+ * The message payload label helps identify the type of event data.
+ */
+export interface CliEventMessage extends Message {
+  type: CliEventMessageType;
+  payload: {
+    label: string;
+    data: any;
+  };
+}
```

Comment:
Thanks @benjamincburns. I'll definitely dig into other message brokers (especially MQTT) in the near future as dashboard's messaging requirements grow. Though I do wonder if adding another dashboard package is better than squishing the types into `@truffle/dashboard-message-bus-common`

> The message bus common package is meant to house logic and types that are common between the message bus service and the message bus client

Something like `@truffle/event-messages` would serve the purpose of providing types to all things message bus, which falls into the domain of the common package?

Ah and about using generics here: I considered it but since the entire message lifecycle for `CliEventMessage` isn't useful at the moment, (this may very well change; right now it's just `abandon`ed after publish, so only `message.payload.data` is useful), I opted to simply pass the payload data to a function (`handleWorkflowCompileResult`) whose params are typed.

Reply:
> Something like @truffle/event-messages would serve the purpose of providing types to all things message bus, which falls into the domain of the common package?

My thinking in the case of a hyptothetical `@truffle/event-messages` was that the only consumers of that package would be event message producers or consumers. `@truffle/dashboard-message-bus-common` is really only meant to contain code that's common to both the message bus client and the message bus service itself, and my thought was that it should only contain logic and/or types that is _strictly required_ for the proper functioning of both. In my ideal world the only direct consumers of `@truffle/dashboard-message-bus-common` would be `@truffle/dashboard-message-bus` and `@truffle/dashboard-message-bus-client`.

Honestly it's probably a mistake on my part to have created the common package without moving the message payload types elsewhere.

Reply:
> Ah and about using generics here: I considered it but since the entire message lifecycle for CliEventMessage isn't useful at the moment, (this may very well change; right now it's just abandoned after publish, so only message.payload.data is useful), I opted to simply pass the payload data to a function (handleWorkflowCompileResult) whose params are typed.

You may want to consider changing the type of `data` from `any` to `unknown`, as that will force an explicit cast. I don't feel strongly about that preference, however. As a pattern it make the code that consumes the `unknown`-typed field more explicit and readable, but in this specific case I don't think that it's likely to add a ton of extra value over `any`.

Reply:
> You may want to consider changing the type of data from any to unknown

Done!